### DISCUSSION TITLE
Updated build avilable

### DISCUSSION THREAD

**Discussion by gitMiguel** [gitMiguel/openhab-addons#8] [General]
Let's continue the discussion [started here as an issue...](https://github.com/gitMiguel/openhab-addons/issues/1)

Here's the download link to newer 3.0 version of this binding:

https://github.com/gitMiguel/openhab-addons/blob/3.1.x/bundles/org.openhab.binding.vallox/org.openhab.binding.vallox-3.1.0-SNAPSHOT.jar

And here's to version 3.3:

https://github.com/gitMiguel/openhab-addons/blob/main/bundles/org.openhab.binding.vallox/org.openhab.binding.vallox-3.3.0-SNAPSHOT.jar

**Comment by flashy02** [comment]
Hi @gitMiguel. Thank you very much. I'll give it a try if I find the time to migrate to OH3.0.  :-)

**Reply by ghost** [reply]
Hi @gitMiguel,

works fine for me with OH v3 so far. Excellent work!

Thx

**Reply by gitMiguel** [reply]
Nice. Let's see what future brings. Happy new year!

**Reply by ghost** [reply]
@gitMiguel Yes, same to you! 

Just curious, as I'm currently woring on setting up my dev environment for OpenHAB: How do you handle the plugin development locally? My Vallox unit is in naturally in "production mode", operating with OpenHAB 3 on a Raspi 4 using an USB serial adapter. For specific debugging purposes I would be able to plugin the adpater to my notebook, but then it's missing from my OpenHAB live system. So for general purpose stuff on such a plugin I would assume, it's a better idea to just deploy modified JARs to the live system while developing. Other option might be to attach the development machine with another USB serial adapter as just one more control unit to the vent unit.

**Reply by gitMiguel** [reply]
For small changes I've just deployed modified .jar and bumped logging to debug or trace and maybe direct logs to separate file. For bigger changes  I run the binding in dev environment and forward the serial port with ser2net/socat to my desktop where I'm developing.

Now as I think of it, a separate usb adapter would imo be the easiest solution.

**Reply by ghost** [reply]
@gitMiguel Well - now I've found your commenta - it was hidden behind a "see more". Good idea with the ser2net. The idea with the separate adapter connected to the Vallox unit is probably something I might try out if I had to find a better adapter for my raspi anyways.

**Reply by ghost** [reply]
@gitMiguel Just a short update:

I've switched my USB adapter from a FTDI FT232RL variant (https://www.amazon.de/DIGITUS-Seriell-Konverter-Einbaubuchse-schwarz/dp/B007VZY4CW/ref=psdc_1626220031_t4_B078X5H8H7) to a CP2102 variant (https://www.amazon.de/gp/product/B078X5H8H7/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1) and now serial connection seems to be stable. System is running without reboot for a week now, and the binding is still working. No "communication error" or such. Hope it stays that way, but so far I'm happy with it!

**Reply by gitMiguel** [reply]
Nice!

**Comment by Achim1601** [comment]
Hi Miika,
First of all, many thanks for working on this!!!
Running a Vallox 90SE, connected via USB/serial, for now just firing some commands via a shell script using the printf command. Very basic, but works for changing the ventilation level.
Now I'm in the process of migrating from OH2 to OH3, installed an OH3 3.0.1 test installation on a separate box. Trying to include the OH3 binding snapshot jar file from above, I've put it into /usr/share/openhab3/addons, but it does not show up in karaf bundle:list.
Would love to test, hence any idea what is wrong?
Many thanks and best regards
Achim

**Reply by Achim1601** [reply]
Sorry, was my fault. Downloaded it again and now it installs. Already thought problem was that ArchLinux installs OH in ~/openhab3 folder structures. As soon as I upgrade my original OH installation (which is connected to the Vallox) I will give it a try and report back.
Cheers
Achim

**Comment by gitMiguel** [comment]
I uploaded a new build. Only minor changes that doesn't matter if your running OH version 3.0.x. Including license header updates, spotless check fixes, and removal of a dependecy to apache ioutils. After this dependency fix this binding should now support 3.1 snapshot versions.

**Comment by Sprocket02** [comment]
With the new version I got the following error (3.1.0.M4 - Milestone Build):

`2021-05-31 10:58:30.387 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/usr/share/openhab/addons/org.openhab.binding.vallox-3.1.0-SNAPSHOT.jar
org.osgi.framework.BundleException: Could not resolve module: org.openhab.binding.vallox [232]
  Unresolved requirement: Import-Package: javax.measure; version="[2.1.0,3.0.0)"
	at org.eclipse.osgi.container.Module.start(Module.java:463) ~[org.eclipse.osgi-3.16.200.jar:?]
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:440) ~[org.eclipse.osgi-3.16.200.jar:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1260) [bundleFile:3.6.8]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1233) [bundleFile:3.6.8]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startAllBundles(DirectoryWatcher.java:1221) [bundleFile:3.6.8]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:515) [bundleFile:3.6.8]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:365) [bundleFile:3.6.8]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:316) [bundleFile:3.6.8]`

**Reply by gitMiguel** [reply]
I've now uploaded a new version that fixes this issue.

**Reply by flashy02** [reply]
Hello everybody, has anybody running the actual jar in an Openhab 3.1 setup with TCP-Connection? At the moment I'm trying to migrate my OH 2.5.12 to OH 3.1 but I've problems to get the vallox binding running. I've copied the actual jar (2021.06.18) in the addons folder and using the old thing/item-file definition, i can see an online Thing on the UI. But every channel I connect, I get only null as value.

**Reply by gitMiguel** [reply]
Do you still have problems @flashy02 ?

**Reply by flashy02** [reply]
@gitMiguel : At the moment everything is fine. I think it was the problem of running two instances (2.5.10 + 3.1) at the same time with the same "display" adress. After shutting down the 2.5.10 instance and reboot of the 3.1 instance it runs fine.

**Reply by flashy02** [reply]
@gitMiguel: would the actual version run under OH 4?

**Reply by gitMiguel** [reply]
@flashy02 please see new markeplace entry: https://community.openhab.org/t/vallox-se-binding-4-0-0-5-0-0/149503
